up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -14,340 +14,129 @@
+// UDiscovery works in heiarchial mode where the local uDiscovery service is the first point of contact
+// and the domain level service (for information about devices within a domain), and then central service
+// is the last point of contact.
+//
+// NOTE: the internal uDiscovery communication protocol (for how data is replicated between the local, domain,
+// and central) is NOT covered in this interface as this is ONLY the client APIs.
 service uDiscovery {
   option (uprotocol.service_name) = "core.udiscovery"; // Service name
   option (uprotocol.service_version_major) = 3;
   option (uprotocol.service_version_minor) = 0;
   option (uprotocol.service_id) = 1;
 
-  // uDiscovery Node Change Notification that sends the Update message
-  option (uprotocol.notification_topic) = {
-    id: 0x8000,
-      name: "NodeChange",
-      message: "Notification"
-  };
 
-    
-
-  // Used by any uProtocol application or service to find service instances
-  // locations. The passed UUri contains valid UEntity, UResource, and UAuthority information
-  // for a query.
-  // What is returned is a list of UUris that match the query.
-  // If lookup fails, the response will be a UStatus with
-  // 2. code.NOT_FOUND: No matching UUris were found
-  // 3. code.INVALID_ARGUMENT: The passed UUri is invalid
-  // 4. code.PERMISSION_DENIED: The caller does not have permission to perform the query
-  rpc LookupUri(LookupUriRequest) returns (LookupUriResponse) {
+  // Find Services. 
+  //
+  // The API is used to find the authority(ies), instance(s), and version(s) of services based on the passed URI.
+  // The only mandatory portion of the URI that must be passed is the ue_id, the rest can be discovered.
+  // When you pass the wildcard for authority_name, the search will be done up to the domain, in order to seach 
+  // in the central discovery database, you must set search_central=true.
+  // The function returns a FindServicesResponse that contains a batch of 1 or more services that match the
+  // search criteria or it will return an error (the method invocation fails) with one of the 
+  // following reasons:
+  // UCode.NOT_FOUND - No service matching the URI was found.
+  // UCode.INVALID_ARGUMENT - The URI passed is invalid
+  // UCode.PERMISSION_DENIED - The caller is not permissed to query for that service
+  //
+  // Examples of URIs that can be passed for searching and their meanings:
+  // UUri::ue_id=1234 - Find all instances and versions of service 1234 for a given device local domain only
+  // UUri::ue_id=1234, UUri::authority_name=* - Find all instances and versions of service 1234 for within a domain
+  // UUri::ue_id=1234, UUri::authority_name=*, search_central=true - Find all instances and versions of service 1234
+  // globally (that is adressable)
+  // UUri::ue_id=11234, UUri::authority_name=*, UUri::version_major=2 - Find a instance 1 of service 1234 and version 2
+  // within a domain.
```

Comment:
This does not seem to be consistent with the semantics defined for URI matching.
FMPOV the query URIs would (should?) be:
1. `up:/FFFF1234/FF/FFFF`
2. `up://*/FFFF1234/FF/FFFF`
3. `up://*/FFFF1234/FF/FFFF`
4. `up://*/11234/2/FFFF`

Reply:
Ok I fixed this issue. Thanks.